src/PowerShellEditorServices/Services/PowerShell/Utility/CommandHelpers.cs

```diff
@@ -168,5 +172,44 @@ internal static class CommandHelpers
 
             return synopsisString;
         }
+
+        /// <summary>
+        /// Gets all aliases found in the runspace
+        /// </summary>
+        /// <param name="executionService"></param>
+        public static async Task<(ConcurrentDictionary<string, List<string>>, ConcurrentDictionary<string, string>)> GetAliasesAsync(IInternalPowerShellExecutionService executionService)
+        {
+            Validate.IsNotNull(nameof(executionService), executionService);
+
+            // TODO: Should we return the caches if they're not empty, or always update?
+            // if (!s_cmdletToAliasCache.IsEmpty || !s_aliasToCmdletCache.IsEmpty)
+            // {
+            //     return (s_cmdletToAliasCache, s_aliasToCmdletCache);
+            // }
+
+            IEnumerable<CommandInfo> aliases = await executionService.ExecuteDelegateAsync<IEnumerable<CommandInfo>>(
+                nameof(GetAliasesAsync),
+                Execution.ExecutionOptions.Default,
+                (pwsh, _) =>
+                {
+                    CommandInvocationIntrinsics invokeCommand = pwsh.Runspace.SessionStateProxy.InvokeCommand;
+                    return invokeCommand.GetCommands("*", CommandTypes.Alias, nameIsPattern: true);
+                },
+                CancellationToken.None).ConfigureAwait(false);
+
+            foreach (AliasInfo aliasInfo in aliases)
+            {
+                List<string> aliasList = new() { aliasInfo.Name };
+
+                s_cmdletToAliasCache.AddOrUpdate(
+                    aliasInfo.Definition,
+                    aliasList,
+                    (_, v) => v.Concat(aliasList).ToList());
```

Comment:
Maybe it's better to use the other overload for `AddOrUpdate` to avoid creating extra `List` instances when multiple alias mapping to the same commands.
```
s_cmdletToAliasCache.AddOrUpdate(
    aliasInfo.Definition,
    (_, arg) => new List<string>(arg),
    (_, v, arg) => { v.Add(arg); return v },
    aliasInfo.Name);
```
Those 2 lambdas will be static delegates.

Reply:
Oh I like that! Thank you.